test/ex_webrtc/dtls_transport_test.exs

```diff
@@ -36,21 +52,20 @@ defmodule ExWebRTC.DTLSTransportTest do
   end
 
   setup do
-    assert {:ok, dtls} = DTLSTransport.start_link([tester: self()], FakeICEAgent)
+    assert {:ok, dtls} = DTLSTransport.start_link(MockICETransport, tester: self())
     assert_receive {:dtls_transport, ^dtls, {:state_change, :new}}
-    ice = DTLSTransport.get_ice_agent(dtls)
-    assert is_pid(ice)
+    {ice_transport, ice_pid} = DTLSTransport.get_ice_transport(dtls)
 
-    %{dtls: dtls, ice: ice}
+    %{dtls: dtls, ice_transport: ice_transport, ice_pid: ice_pid}
   end
 
-  test "forwards non-data ICE messages", %{ice: ice} do
+  test "forwards non-data ICE messages", %{ice_transport: ice_transport, ice_pid: ice_pid} do
     message = :connected
 
-    FakeICEAgent.send_dtls(ice, message)
+    ice_transport.send_dtls(ice_pid, message)
```

Comment:
Well, it's not like you're going to use anything else other than `MockICETransport` as an `ICETransport`, but you went all the way through with the mocking game, I guess.

Reply:
Not really, tbh it's my first time playing with mocks but I did it to be able to easily change MockICETransport to something other or to rename MockICETransport